Source/WebCore/bindings/scripts/CodeGeneratorJS.pm

```diff
@@ -5376,17 +5376,11 @@ extern "C" { extern void (*const ${vtableRefWin}[])(); }
 #else
 extern "C" { extern void* ${vtableNameGnu}[]; }
 #endif
-#endif
-
-END
-
-        push(@implContent, "JSC::JSValue toJSNewlyCreated(JSC::JSGlobalObject*, JSDOMGlobalObject* globalObject, Ref<$implType>&& impl)\n");
-        push(@implContent, "{\n");
-        push(@implContent, <<END) if $vtableNameGnu;
-
-    if constexpr (std::is_polymorphic_v<${implType}>) {
-#if ENABLE(BINDING_INTEGRITY)
-        const void* actualVTablePointer = getVTablePointer(impl.ptr());
+    // Force the type being checked to be a dependent type so we can rely on
+    // constexpr if not causing errors when evaluated.
+template<typename T, typename = std::enable_if_t<std::is_same_v<T, ${implType}>, void>> static inline void verifyVTable(const ${implType}* ptr) {
+    if constexpr (std::is_polymorphic_v<T>) {
+        const void* actualVTablePointer = getVTablePointer(ptr);
```

Comment:
Couldn't we instead just make a getVTablePointer for non-polymorphic types using requires or std::enable_if_t that will never actually be instantiated?

Reply:
No because you hit the same problem, which is that there is no dependent type or value involved, so the body of the if must be valid, but if getVTablePointer is gated by enable_if there is no valid getVTablePointer and the code errors out.

That said it might be worth adding the gate anyway, because the existing clang behavior already errors out in this case (my guess is the existing behavior is a result of incorrectly deferring instantiation of getVTablePointer inside the constexpr - putting an enable_if guard on getVTablePointer means that no matching function can be resolved at all so it errors out on buggy compilers as well)

Reply:
oh sorry, I see what you mean - add a getVTablePointer that is gated on !polymorphic, relying on it being considered more tightly constrained and therefore preferred over the generic version. We could make it reference a non existing symbol as well so it was guaranteed to not link if a call ever does get emitted.